turbopack/crates/turbopack-ecmascript/src/tree_shake/graph.rs

```diff
@@ -273,12 +279,23 @@ impl DepGraph {
                 for var in item.var_decls.iter() {
                     declarator.entry(var.clone()).or_insert_with(|| ix as u32);
                 }
+
+                if let Some(export) = &item.export {
+                    exporter.insert(export.clone(), ix as u32);
+                }
             }
         }
 
         let mut mangle_count = 0;
         let mut mangled_vars = FxHashMap::default();
         let mut mangle = |var: &Id| {
+            // For react hooks.
+            // We need to preserve `use*` to make error reporting logic work.
+            // We catch `useXxx is not a function`, and report it with a hint about 'use client'.
+            if var.0.starts_with("use") {
+                return var.0.clone();
+            }
```

Comment:
That's only needed in development?

I don't want to introduce a production inefficiency because of some development feature. Maybe we can add a toggle to disable mangling in development (if tree shaking is used there, which isn't by default). So we can get rid of this special case for react...